SetItemListenerOnReconnectTest.testListenersNonSmartRoutingMultipleServer

```
java.lang.AssertionError: Received event count is 0 but it is expected to stay at 10 expected:<10> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest$9$1.run(AbstractListenersOnReconnectTest.java:528)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueAllTheTime(HazelcastTestSupport.java:951)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest$9.run(AbstractListenersOnReconnectTest.java:523)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:967)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:984)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.validateListenerFunctionality(AbstractListenersOnReconnectTest.java:510)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.validateRegistrationsAndListenerFunctionality(AbstractListenersOnReconnectTest.java:473)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.testListenersTerminateOwnerNode(AbstractListenersOnReconnectTest.java:458)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.testListenersNonSmartRoutingMultipleServer(AbstractListenersOnReconnectTest.java:409)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:89)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:81)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast-client/769/testReport/junit/com.hazelcast.client.listeners/SetItemListenerOnReconnectTest/testListenersNonSmartRoutingMultipleServer/

--- a/hazelcast-client/src/test/java/com/hazelcast/client/listeners/AbstractListenersOnReconnectTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/listeners/AbstractListenersOnReconnectTest.java
@@ -519,7 +519,7 @@ public abstract class AbstractListenersOnReconnectTest extends ClientTestSupport
                 for (String event : events) {
                     produceEvent(event);
                 }
-
+                assertOpenEventually(eventsLatch);
                 assertTrueAllTheTime(new AssertTask() {
                     @Override
                     public void run()

